slither/detectors/statements/assembly.py

```diff
@@ -0,0 +1,58 @@
+"""
+Module detecting usage of inline assembly
+"""
+
+from slither.detectors.abstract_detector import AbstractDetector, DetectorClassification
+
+
+class Assembly(AbstractDetector):
+    """
+    Detect usage of inline assembly
+    """
+
+    ARGUMENT = 'assembly'
+    HELP = 'assembly usage'
+    IMPACT = DetectorClassification.INFORMATIONAL
+    CONFIDENCE = DetectorClassification.HIGH
+
+    @staticmethod
+    def _contains_inline_assembly_use(node):
+        """
+             Check if the node contains ASSEMBLY type
+        Returns:
+            (bool)
+        """
+        return node.type == 0x14
```

Comment:
Consider comparing to [NodeType.Assembly](https://github.com/trailofbits/slither/blob/265a7d6a177deb8fd04277694d24e48e303782ed/slither/core/cfg/node.py#L31) for maintainability

Reply:
Doing that asap.

Reply:
Done.